cheatsheets/Java_Security_Cheat_Sheet.md

```diff
@@ -0,0 +1,685 @@
+not by preencoding all the log messages with HTML encoding as such log entries may be used/viewed in many
+other log viewing/analysis tools that don't expect the log data to be pre-HTML encoded.
+```
+
+- [LOG4J Configuration](https://logging.apache.org/log4j/2.x/manual/configuration.html)
+- [LOG4J Appender](https://logging.apache.org/log4j/2.x/manual/appenders.html)
+- [Log Forging](https://github.com/javabeanz/owasp-security-logging/wiki/Log-Forging) - See the Logback section about the `CRLFConverter` this library provides.
+- [Usage of OWASP Security Logging with Logback](https://github.com/javabeanz/owasp-security-logging/wiki/Usage-with-Logback)
+
+## Cryptography
+
+### General cryptography guidance
+
+- **Never, ever write your own cryptographic functions.**
+- Make sure your application or protocol can easily support a future change of cryptographic algorithms.
+- Use your package manager whereever possible to keep all of your packages up to date. Watch the updates on your development setup, and plan updates to your applications accordingly.
+
+### Encryption for storage
+
+Follow the algorithm guidance in the [OWASP Cryptographic Storage Cheat Sheet](https://cheatsheetseries.owasp.org/cheatsheets/Cryptographic_Storage_Cheat_Sheet.html#algorithms).
+
+The following code snippet shows an example of using AES-GCM to perform encryption/decryption of data.
+
+```java
+import java.nio.charset.StandardCharsets;
+import java.security.SecureRandom;
+import javax.crypto.spec.*;
+import javax.crypto.*;
+import java.util.Base64;
+
+
+// AesGcmSimpleTest
+class Main {
+
+    public static void main(String[] args) throws Exception {
+        // Key of 32 bytes / 256 bits for AES
+        KeyGenerator keyGen = KeyGenerator.getInstance(AesGcmSimple.ALGORITHM);
+        keyGen.init(AesGcmSimple.KEY_SIZE, new SecureRandom());
+        SecretKey secretKey = keyGen.generateKey();
+
+        // Nonce of 12 bytes / 96 bits and this size should always be used.
+        byte[] nonce = new byte[AesGcmSimple.IV_LENGTH];
+        SecureRandom random = new SecureRandom();
+        random.nextBytes(nonce);
+
+        var message = "This message to be encrypted";
+        System.out.println(message);
+
+        // Encrypt the message
+        byte[] cipherText = AesGcmSimple.Encrypt(message, nonce, secretKey);
+        System.out.println(Base64.getEncoder().encodeToString(cipherText));
+
+        // Decrypt the message
+        var message2 = AesGcmSimple.Decrypt(cipherText, nonce, secretKey);
+        System.out.println(message2);
+    }
+}
+
+class AesGcmSimple {
+
+    public static final String ALGORITHM = "AES";
+    public static final String CIPHER_ALGORITHM = "AES/GCM/NoPadding";
+    public static final int KEY_SIZE = 256;
+    public static final int TAG_LENGTH = 128;
+    public static final int IV_LENGTH = 12;
+
+    public static byte[] Encrypt(String plaintext, byte[] nonce, SecretKey secretKey) throws Exception {

```

Comment:
Was this copied from some .NET C# code? By convention, Java methods start with lower case.